feature/login/src/main/java/com/epmedu/animeal/login/profile/presentation/FinishProfileViewModel.kt

```diff
@@ -0,0 +1,161 @@
+package com.epmedu.animeal.login.profile.presentation
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.text.AnnotatedString
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.epmedu.animeal.foundation.input.BirthDateFormatTransformation
+import com.epmedu.animeal.login.profile.data.model.Profile
+import com.epmedu.animeal.login.profile.data.repository.ProfileRepository
+import com.epmedu.animeal.login.profile.domain.BirthDateValidator
+import com.epmedu.animeal.login.profile.domain.EmailValidator
+import com.epmedu.animeal.login.profile.domain.FirstnameValidator
+import com.epmedu.animeal.login.profile.domain.SurnameValidator
+import com.epmedu.animeal.login.profile.domain.ValidationResult
+import com.epmedu.animeal.login.profile.presentation.ui.ProfileEvent
+import com.epmedu.animeal.login.profile.presentation.ui.ProfileState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.receiveAsFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+internal class FinishProfileViewModel @Inject constructor(
+    private val profileRepository: ProfileRepository
+) : ViewModel() {
+    private val nameValidator: FirstnameValidator = FirstnameValidator
+    private val surnameValidator: SurnameValidator = SurnameValidator
+    private val emailValidator: EmailValidator = EmailValidator
+    private val birthDateValidator: BirthDateValidator = BirthDateValidator
+
+    var state by mutableStateOf(ProfileState())
+    private val validationEventChannel = Channel<ValidationEvent>()
+    val validationEvents = validationEventChannel.receiveAsFlow()
```

Comment:
As I remember Channel deprecated in favour of SharedFlow
Could you please migrate?

Reply:
Done